MicroPython/src/cloud/KAA_cloud.py

```diff
@@ -0,0 +1,190 @@
+import logging
+import random
+from random import randint
+
+import machine
+import ujson
+from common import config, utils
+from communication import wirerless_connection_controller
+from controller.main_controller_event import (MainControllerEvent,
+                                              MainControllerEventType)
+from utime import time
+
+from cloud.cloud_interface import CloudProvider
+
+
+class KAA_cloud(CloudProvider):
+    def __init__(self) -> None:
+        self.publish_success_topic = config.cfg.kaa_topic + '/status'
+        self.publish_error_topic = config.cfg.kaa_topic + '/error'
+
+    def receive_message(self, topic, msg) -> None:
+        """
+        Callback method for MQTT client
+        :param topic: Topic of the message received encoded as bytes
+        :param msg: Message received encoded as bytes
+        :return: None 
+        """
+        logging.debug('cloud/Kaa_cloud.py/receive_message()')
+        topic = topic.decode()
+
+        # Check if msg is in json format, if not decode as str
+        if b'{' in msg and b'}' in msg:
+            msg = ujson.loads(msg)
+        else:
+            msg = msg.decode()
+
+        if topic == self.publish_success_topic:
+            if msg == '':
+                logging.info('Operation successful\n')
+            else:
+                logging.info('Operation successful with return code: {}\n'.format(msg))
+        elif topic == self.publish_error_topic:
+            status_code = msg['statusCode']
+            reason = msg['reasonPhrase']
+            logging.info('Operation failed with error code: {} - reason: {}\n'.format(
+                status_code, reason
+            ))
+        else:
+            logging.info('On topic: {} received msg: {}'.format(topic, msg))
+
+    def device_configuration(self, data: dict) -> int:
```

Comment:
is this exactly the same method as in AWS cloud?

Reply:
It's not, as the one in AWS code needs to register aws_thing, and the one in KAA needs to adjust (and load) some config fields related to communication with KaaIoT